delivery_postlogistics/models/stock_picking.py

```diff
@@ -0,0 +1,238 @@
+            .with_context(context_attachment)
+            .create(data)
+        )
+
+    def _set_a_default_package(self):
+        """ Pickings using this module must have a package
+            If not this method put it one silently
+        """
+        for picking in self:
+            move_lines = picking.move_line_ids.filtered(
+                lambda s: not (s.package_id or s.result_package_id)
+            )
+            if move_lines:
+                default_packaging = (
+                    picking.carrier_id.postlogistics_default_packaging_id
+                )
+                package = self.env["stock.quant.package"].create(
+                    {
+                        "packaging_id": default_packaging
+                        and default_packaging.id
+                        or False
+                    }
+                )
+                move_lines.write({"result_package_id": package.id})
+
+    def postlogistics_cod_amount(self):
+        """ Return the Postlogistic Cash on Delivery amount of a picking
+
+        If the picking delivers the whole sales order, we use the total
+        amount of the sales order.
+
+        Otherwise, we don't know the value of each picking so we raise
+        an error.  The user has to create packages with the cash on
+        delivery price on each package.
+        """
+        self.ensure_one()
+        order = self.sale_id
+        if not order:
+            return 0.0
+        if len(order) > 1:
+            raise exceptions.Warning(
+                _(
+                    "The cash on delivery amount must be manually specified "
+                    "on the packages when a package contains products "
+                    "from different sales orders."
+                )
+            )
+        order_moves = order.mapped("order_line.procurement_ids.move_ids")
+        picking_moves = self.move_lines
+        # check if the package delivers the whole sales order
+        if order_moves != picking_moves:
+            raise exceptions.Warning(
+                _(
+                    "The cash on delivery amount must be manually specified "
+                    "on the packages when a sales order is delivered "
+                    "in several delivery orders."
+                )
+            )
+        return order.amount_total
+
+    def write_tracking_number_label(self, label_result, packages):
+        """
+        If there are no pack defined, write tracking_number on picking
+        otherwise, write it on parcel_tracking field of each pack.
+        Note we can receive multiple labels for a same package
+        """
+
+        def info_from_label(label):
+            tracking_number = label["tracking_number"]
+            data = base64.b64decode(label["binary"])
+            if label["file_type"] == "zpl2":
+                data = base64.b64encode(
+                    base64.b64decode(data)
+                    .decode("cp437")
+                    .replace("^XA", "^XA^CI28")
```

Comment:
We apparently also need to enable the right font on the printer.
Could you make this entire replace configurable on the carrier.
If boolean "Do not patch ZPL to enable UTF-8" is False (default), then apply the replace
If True, then do not replace
```suggestion
                    .replace("^XA", "^XA^CW0,E:TT0003M_.TTF^XZ^XA^CI28")
```

Reply:
Maybe instead of a boolean on the carrier, we could put the full replacement string. Then it's not hardcoded anymore

Reply:
Add in https://github.com/OCA/delivery-carrier/pull/323/commits/430e52d51d740eb55e962336dd11056827aba033

Reply:
Thanks